src/main/scala/util/RunUtils.scala

```diff
@@ -455,6 +463,70 @@ object IRTransform {
     ctx.program.setModifies(specModifies)
   }
 
+  // identify calls to pthread_create
+  // use analysis result to determine the third parameter's value (the function pointer)
+  // split off that procedure into new thread
+  // do reachability analysis
+  // also need a bit in the IR where it creates separate files
+  def splitThreads(program: Program,
+                   pointsTo: Map[RegisterVariableWrapper, Set[RegisterVariableWrapper | MemoryRegion]],
+                   regionContents: Map[MemoryRegion, Set[BitVecLiteral | MemoryRegion]],
+                   reachingDefs: Map[CFGPosition, (Map[Variable, Set[LocalAssign]], Map[Variable, Set[LocalAssign]])]
+                  ): Unit = {
+    
+    // iterate over all commands - if call is to pthread_create, look up?
+    for (p <- program.procedures) {
+      for (b <- p.blocks) {
+        b.jump match {
+          case d: DirectCall if d.target.name == "pthread_create" =>
+
+            // R2 should hold the function pointer of the function that begins the thread
+            // look up R2 value using points to results
+            val R2 = Register("R2", BitVecType(64))
+            val b = reachingDefs(d)
+            val R2Wrapper = RegisterVariableWrapper(R2, getDefinition(R2, d, reachingDefs))
+            val threadTargets = pointsTo(R2Wrapper)
+
+            if (threadTargets.size > 1) {
+              // currently can't handle case where the thread created is ambiguous
+              throw Exception("can't handle thread creation with more than one possible target")
+            }
```

Comment:
Since all we currently do is output extra programs to verify we can probably just attempt to resolve all the targets and add them as separate threads.